chronicle-queue/src/test/scala/akka/stream/alpakka/chroniclequeue/ChronicleQueueSpec.scala

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+// ORIGINAL LICENCE
+/*
+ *  Copyright 2017 PayPal
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package akka.stream.alpakka.chroniclequeue
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.stream.scaladsl.{Broadcast, Flow, GraphDSL, Keep, RunnableGraph, Sink, Source}
+import akka.stream.{AbruptTerminationException, ActorMaterializer, ClosedShape}
+import akka.util.ByteString
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.Eventually
+import org.scalatest.time.{Seconds, Span}
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+import scala.concurrent.{Await, Promise}
+import scala.reflect._
+
+import akka.stream.alpakka.chroniclequeue.scaladsl._
+import Timeouts._
+
+object ChronicleQueueSpec {
+  val testConfig = ConfigFactory.parseString("""
+      |akka.actor.default-dispatcher {
+      |  executor = "thread-pool-executor"
+      |  thread-pool-executor {
+      |    core-pool-size-min = 1
+      |    core-pool-size-factor = 2.0
+      |    core-pool-size-max = 10
+      |  }
+      |}
```

Comment:
Use `affinity-pool` is a much better choice here.

Reply:
I did this for compatibility with akka 2.4 reverted to affinity pool here